src/storage/hub.ts

```diff
@@ -51,7 +52,8 @@ export async function uploadToGaiaHub(
     }
   )
   if (!response.ok) {
-    throw new Error('Error when uploading to Gaia hub')
+    const responseDescription = await getResponseDescription(response)
```

Comment:
I know this is still on your PR TODO list -- but here is the list of Gaia hub upload error responses: https://github.com/blockstack/gaia/blob/ff1cd245e4df7244e8c6e32f08e95ec2aba9834e/hub/src/server/http.ts#L68-L79

I think we will probably want corresponding error codes and class types that match those.

Reply:
Done 👍

Reply:
Looking good! Couple minor change requests:
* The Gaia has a [`413 PayloadTooLarge`](https://github.com/blockstack/gaia/blob/ef0771833cd56200e08e18631e62467f24e086af/hub/src/server/http.ts#L78) error code _(it was recently introduced, you didn't miss it)_. 
* The [`delete` file endpoint](https://github.com/blockstack/gaia/blob/ef0771833cd56200e08e18631e62467f24e086af/hub/src/server/http.ts#L120-L131) can also return many of these same errors, in addition to a [specific 404 error](https://github.com/blockstack/gaia/blob/ef0771833cd56200e08e18631e62467f24e086af/hub/src/server/http.ts#L127). The blockstack.js function for that is located right underneath this one, see `function deleteFromGaiaHub`.. 
* Same thing for the [`list-files` endpoint](https://github.com/blockstack/gaia/blob/ef0771833cd56200e08e18631e62467f24e086af/hub/src/server/http.ts#L172). See `function listFilesLoop`.. which is in `src/storage/index.ts`.

All these fetch requests to the Gaia server should probably use a common error handling function, i.e. all `fetch` requests involving `GaiaHubConfig.server`.

Reply:
Done--`getBlockstackErrorFromResponse` in utils.ts is the central place to process gaia hub errors. Thanks!